src/region_growing.jl

```diff
@@ -0,0 +1,163 @@
+
+# An implementation of the Improved Seeded Region Growing Algorithm in Julia
+#
+# Citation:
+# Albert Mehnert, Paul Jackaway (1997), "An improved seeded region growing algorithm"
+# http://www.ee.bgu.ac.il/~itzik/IP5211/Other/Projects/P19_Seeded%20region%20growing.pdf
+
+function seeded_region_growing{CT<:Colorant, N}(img::AbstractArray{CT,N}, seeds::AbstractVector{Tuple{CartesianIndex{N},Int}},
+    kernel_dim::Vector{Int} = [3 for i in 1:N], diff_fn::Function = (c1,c2)->(sqrt(sum(abs2,wrapping_type(CT)(c1)-wrapping_type(CT)(c2)))))
+    length(kernel_dim) == N || error("Dimension count of image and kernel_dim do not match")
+    for dim in kernel_dim
+        dim > 0 || error("Dimensions of the kernel must be positive")
+        isodd(dim) || error("Dimensions of the kernel must be odd")
+    end
+    pt = CartesianIndex([floor(Int, dim/2) for dim in kernel_dim]...)
```

Comment:
Trivial, but you can use `dim ÷ 2` (`\divTAB`) or `dim >> 1`. The former is probably more readable.